jwst/assign_wcs/nirspec.py

```diff
@@ -256,29 +256,97 @@ def get_open_fixed_slits(input_model):
     return slits
 
 
-def get_open_msa_slits(msa_status):
+def get_open_msa_slits(msa_file, msa_metadata_id=88):
+    """
+    The msa_file should be an open fits file from which the
+    shutter information will be read.
+
+    The file is expected to contain data (tuples) with the following fields:
+
+        ('slitlet_id', '>i2'),
+        ('msa_metadata_id', '>i2'),
+        ('shutter_quadrant', '>i2'),
+        ('shutter_row', '>i2'),
+        ('shutter_column', '>i2'),
+        ('source_id', '>i2'),
+        ('background', 'S1'),
+        ('shutter_state', 'S6'),
+        ('estimated_source_in_shutter_x', '>f4'),
+        ('estimated_source_in_shutter_y', '>f4')])
+
+    For example, something like:
+        (12, 2, 4, 251, 22, 1, 'Y', 'OPEN', nan, nan),
+
+    parameters
+        msa_file: Open fits file that contains the configuration information
+        msa_metadata_id: The MSA meta id that we are going to work on...
     """
-    Returns slit_id for all open slits.
-
-    Parameters
-    ----------
-    msa_status : str
-        Name of MSA status file.
 
-    Returns
-    -------
-    msa_slits_id : ndarray
-        An array of slit ids.
-        A slit_id is a tuple of (quadrant_number, slit_number).
-
-    """
-    with fits.open(msa_status) as fmsa:
-        msa_slits_id = []
-        for i in range(1, 5):
-            quadrant = fmsa[i].data
-            open_shutters = quadrant[quadrant['status'].nonzero()]
-            msa_slits_id.extend([(i, n) for n in open_shutters['NO']])
-    return np.array(msa_slits_id)
+    # If they passed in a string then we shall assume it is the filename
+    # of the configuration file.
+    if isinstance(msa_file, str):
+        try:
+            msa_file = fits.open(msa_file)
+        except Exception as e:
+            log.error(__file__ + ': Could not find/open file {}'.format(e))
+            raise Exception(e)
+
+    # Get the configuration header from teh _msa.fits file.  The EXTNAME should be 'SHUTTER_INFO' but in the
+    # old data files it might be 'MSA SHUTTER INFO TABLE'.
+    try:
+        msa_conf = [h for h in msa_file if h.header['EXTNAME'] in ['MSA SHUTTER INFO TABLE', 'SHUTTER_INFO']]
+        msa_conf = msa_conf[0]
+    except IndexError as e:
+        log.error('get_open_msa_slits: Could not find SHUTTER_INFO EXTNAME in {}'.format(msa_file))
+
+    # First we are going to filter the msa_file data on the msa_metadata_id
+    # as that is all we are interested in for this function.
+    msa_data = [x for x in msa_conf.data if x[1] == msa_metadata_id]
```

Comment:
I'd prefer if we use filed names instead of index the data as it is much more readable.
```
msa_data = [x for x in msa_conf.data if x['msa_metadata_id'] == msa_metadata_id]
```